ScreenControl_Unity/Assets/ScreenControl/Client/Scripts/Plugin System/InputActionPlugin.cs

```diff
@@ -0,0 +1,36 @@
+﻿using UnityEngine;
+
+namespace Ultraleap.ScreenControl.Client
+{
+    // Class: InputActionPlugin
+    // A base class for Plugins which are used by the <InputActionManager> to manipulate the
+    // <ClientInputAction> data received from the Service.
+    public class InputActionPlugin : MonoBehaviour
+    {
+        // Variable: TransmitInputAction
+        // An event for transmitting <ClientInputActions> as they pass through this plugin.
+        // This can be used to access the data as it is used by a specific plugin, as to intercept
+        // the full cycle of plugins that the <InputActionManager> references.
+        public event InputActionManager.ClientInputActionEvent TransmitInputAction;
```

Comment:
Do we need to _require_ this event on every plugin? We're using `RunPlugin` directly from the `InputActionManager`, so we don't directly _need_ it. Though I can see us listening directly to events from specific plugins I'm not sure it's needed in the base.

Reply:
We don't _need_ to, but my thinking is that then any plugin has an event that can be listened to and the standard behavior should be to call the base to trigger that.

I think it becomes hidden as a concept if we don't include it here and if nothing listens it comes at very little (or no?) cost